### PR TITLE
Onboard new RPs for autogeneration by default

### PR DESCRIPTION
**Pull request by anthony-c-martin**

### CODE REVIEW

File: generator/autogenlist.ts [hunk 1]
```diff
@@ -8,8 +8,224 @@ import { postProcessor as policyProcessor } from './processors/Microsoft.Authori
...
+        basePath: 'videoanalyzer/resource-manager',
+        namespace: 'Microsoft.Media',
+        disabledForAutogen: true,
+    },
+    {
+        basePath: 'webpubsub/resource-manager',
+        namespace: 'Microsoft.SignalRService',
+        disabledForAutogen: true,
+    },
+];
```
Comment (anthony-c-martin) [inline generator/autogenlist.ts:224]: This list of a snapshot of the `list-basepaths` output - to avoid changing the list of RPs currently opted in/out.

**Review by apclouds — APPROVED** [review]

File: generator/autogenlist.ts [hunk 2]
```diff
@@ -268,10 +484,10 @@ const autoGenList: AutoGenConfig[] = [
         basePath: 'EnterpriseKnowledgeGraph/resource-manager',
         namespace: 'Microsoft.EnterpriseKnowledgeGraph',
     },
-    /*{ NOTE(jcotillo): Temporally removing this RP - latest swagger contains an unsupported type by the schema generator tool
```
Comment (jorgecotillo) [inline generator/autogenlist.ts:271]: >latest swagger contains an unsupported type [](http://example.com/codeflow?start=54&length=43)

I recall we had issues when generating this schema, would this break our schema generation?

**Comment by jorgecotillo** [comment]
> export const blocklist = [

Something I quite didn't get is what is the criteria to decide which RPs go here, which ones don't, do we just wait until the schema generation fails and we blocklist them?

---
Refers to: generator/constants.ts:38 in d1f8696. [](commit_id = d1f8696736f3fa1c9a7ee9d63b42d3306a353a0d, deletion_comment = False)

File: generator/autogenlist.ts [hunk 3]
```diff
@@ -8,8 +8,224 @@ import { postProcessor as policyProcessor } from './processors/Microsoft.Authori
 import { postProcessor as securityInsightsPostProcessor } from './processors/Microsoft.SecurityInsights';
 import { lowerCaseEquals } from './utils';
 
+// Going forwards, providers are onboarded by default.
+// The providers listed here are the only ones **not** onboarded.
+const disabledProviders: AutoGenConfig[] = [
```
Comment (jorgecotillo) [inline generator/autogenlist.ts:13]: >disabledProviders [](http://example.com/codeflow?start=6&length=17)

What is the difference between this and blocklist?

Comment (anthony-c-martin) [reply generator/autogenlist.ts:13]: Good question. I see the blocklist more like "these paths should be totally ignored, and do not apply for schema gen", whereas the config here is "these RPs have not yet been onboarded, but will be in the future". Let me add some comments and possible rename variables to try and make this clearer.

**Pull request merged**